src/providers/krb5/krb5_ccache.c

```diff
@@ -529,6 +529,7 @@ errno_t sss_krb5_cc_verify_ccache(const char *ccname, uid_t uid, gid_t gid,
 
     mcred.client = princ;
     mcred.server = tgt_princ;
+    /* coverity[store_truncates_time_t] */
```

Comment:
Well, it seems Kerberos is not planing on passing to a 64-bit timestamp. Instead, it would seem they plan on making signed this unsigned value.

See https://web.mit.edu/kerberos/krb5-latest/doc/appdev/y2038.html and https://web.mit.edu/kerberos/krb5-latest/doc/appdev/refs/types/krb5_timestamp.html#c.krb5_timestamp

Considering their recommendations, should we instead fix our code like this:

```*((uint32_t *) &mcred.times.endtime) = (uint32_t) (time(NULL) & 0xFFFFFFFF)```

Reply:
Hi,

if I understand it correctly just

    mcred.times.endtime = (time(NULL) & 0xFFFFFFFF);

should be sufficient because the signed/unsigned state only becomes important when interpreting the value of `mcred.times.endtime`. Since the 32 least significant bits in the 64bit time_t value correspond to an unsigned 32bit value just setting the bits is sufficient.

bye,
Sumit

Reply:
Done.